File: app/src/main/java/com/rachelleignacio/listbucket/domain/interactors/base/Interactor.java
package com.rachelleignacio.listbucket.domain.interactors.base;

/**
 * Created by rachelleignacio on 3/6/17.
 * Interface containing the main method that starts an interactor, making sure the operation is done
 * on a background thread.
 */
public interface Interactor {
    void execute();
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/domain/interactors/base/Interactor.kt
package com.rachelleignacio.listbucket.domain.interactors.base

/**
 * Created by rachelleignacio on 8/9/17.
 * Interface containing the main method that starts an interactor, making sure the operation is done
 * on a background thread.
 */
interface Interactor {
    fun execute()
}